cmdstanpy/model.py

```diff
@@ -856,58 +865,56 @@ def sample(
-                                msg = (
-                                    'Cannot import tqdm.tqdm_notebook.\n'
-                                    'Functionality is only supported on the '
-                                    'Jupyter Notebook and compatible platforms'
-                                    '.\nPlease follow the instructions in '
-                                    'https://github.com/tqdm/tqdm/issues/394#'
-                                    'issuecomment-384743637 and remember to '
-                                    'stop & start your jupyter server.'
-                                )
-                                get_logger().warning(msg)
-                                tqdm_pbar = tqdm.tqdm
-                        else:
-                            tqdm_pbar = tqdm.tqdm
-                        # enable dynamic_ncols for advanced users
-                        # currently hidden feature
-                        dynamic_ncols_raw = os.environ.get(
-                            'TQDM_DYNAMIC_NCOLS', 'False'
-                        )
-                        if dynamic_ncols_raw.lower() in ['0', 'false']:
-                            dynamic_ncols = False
-                        else:
-                            dynamic_ncols = True
-                        pbar = tqdm_pbar(
-                            desc='Chain {} - warmup'.format(i + 1),
-                            position=i,
-                            total=1,  # Will set total from Stan's output
-                            dynamic_ncols=dynamic_ncols,
-                        )
-                        all_pbars.append(pbar)
-                    executor.submit(self._run_cmdstan, runset, i, pbar)
+            # progress reporting
+            iter_total = 0
+            if iter_warmup is None:
+                iter_total += _CMDSTAN_WARMUP
+            else:
+                iter_total += iter_warmup
+            if iter_sampling is None:
+                iter_total += _CMDSTAN_SAMPLING
+            else:
+                iter_total += iter_sampling
+            if refresh is None:
+                refresh = _CMDSTAN_REFRESH
+            iter_total = iter_total // refresh + 2
 
-            # Closing all progress bars
-            for pbar in all_pbars:
-                pbar.close()
             if show_progress:
-                # re-enable logger for console
-                get_logger().propagate = True
+                try:
+                    # pylint: disable=unused-import
+                    from tqdm.auto import tqdm  # type: ignore # noqa
+                except ImportError:
+                    get_logger().warning(
+                        'Package tqdm not installed, cannot show progress '
+                        'information. Please install tqdm with '
+                        "'pip install tqdm'"
+                    )
+                    show_progress = False
+                    show_console = True
```

Comment:
I don't think turning on show_console if show_progress fails is a good idea, we should just emit the warning (I might either want progress bars or no output, which this currently doesn't allow).

It would be nice if we had the makevar equivalent of ?= for this, but currently this would even overwrite if I passed `show_console=False`

Reply:
got rid of line 895